packages/lockfile-file/src/lockfileName.ts

```diff
@@ -0,0 +1,25 @@
+import { WANTED_LOCKFILE } from '@pnpm/constants'
+import { getCurrentBranch } from '@pnpm/git-utils'
+
+export interface GetWantedLockfileNameOptions {
+  useGitBranchLockfile?: boolean
+  mergeGitBranchLockfiles?: boolean
+}
+
+export async function getWantedLockfileName (opts: GetWantedLockfileNameOptions = { useGitBranchLockfile: false, mergeGitBranchLockfiles: false }) {
+  if (opts.useGitBranchLockfile && !opts.mergeGitBranchLockfiles) {
+    const currentBranchName = await getCurrentBranch()
+    if (currentBranchName) {
+      return WANTED_LOCKFILE.replace('.yaml', `.${stringifyBranchName(currentBranchName)}.yaml`)
```

Comment:
what about when the current branch is main or master? Or any other default branch?